### PR TITLE
Local auth

### PR DESCRIPTION
**Pull request by medgyes**
Known issues, todos:

- [x] root@local user should be protected from changing the email address
- [x] user deletion
- [x] ui tests

### CODE REVIEW

File: mitzu/webapp/auth/authorizer.py [hunk 1]
```diff
@@ -304,3 +308,26 @@ def after_request(resp: flask.Response):
     def is_request_authorized(self, request: flask.Request) -> bool:
         auth_token = request.cookies.get(self._config.token_cookie_name)
         return auth_token is not None and self._validate_token(auth_token) is not None
+
+    def login_local_user(self, email: str, password: str) -> Optional[str]:
+        if self._config.user_service is None:
+            msg = "User service is not set for local auth"
```
Comment (IstvanM) [inline mitzu/webapp/auth/authorizer.py:314]: Imho this should be an exception here.

File: mitzu/webapp/auth/authorizer.py [hunk 2]
```diff
@@ -304,3 +308,26 @@ def after_request(resp: flask.Response):
...
+            LOGGER.warning(msg)
+            return msg
+
+        user = self._config.user_service.get_user_by_email_and_password(email, password)
+        if user is None:
+            return "Bad credentials"
```
Comment (IstvanM) [inline mitzu/webapp/auth/authorizer.py:320]: If above we have an exception the return type can be bool.

File: mitzu/webapp/dependencies.py [hunk 3]
```diff
@@ -20,29 +21,48 @@ class Dependencies:
     storage: S.MitzuStorage
     queue: C.MitzuCache
     cache: C.MitzuCache
+    users: Optional[U.UserService] = field(default_factory=lambda: None)
```
Comment (IstvanM) [inline mitzu/webapp/dependencies.py:24]: This can be simply `users: Optional[U.UserService] = None`

Comment (IstvanM) [reply mitzu/webapp/dependencies.py:24]: None is not a complex type. Only complex types like Classes, Lists, Dicts need to be initialized with a `field`.

File: mitzu/webapp/service/user.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import string
+import uuid
+import hashlib
+import mitzu.webapp.cache as C
+
+
+ROOT_USER_EMAIL = "root@local"
+
+
+@dataclass
```
Comment (IstvanM) [inline mitzu/webapp/service/user.py:13]: should this be frozen?

File: Makefile [hunk 5]
```diff
@@ -83,13 +83,20 @@ serve_google_sso:
...
 	GOOGLE_PROJECT_ID="mitzu-test" \
 	GOOGLE_REDIRECT_URL="http://localhost:8082/auth/oauth" \
 	$(POETRY) run gunicorn -b 0.0.0.0:8082 app:server --reload
 
+serve_local_auth:
+	cd release/app/ && \
+	LOG_LEVEL=WARN \
+	LOG_HANDLER=stdout \
+	AUTH_BACKEND="local" \
+	AUTH_ROOT_PASSWORD="test" \
```
Comment (IstvanM) [inline Makefile:98]: Should we also define the AUTH_ROOT_EMAIL in configs?

File: mitzu/webapp/service/user.py (hunk 4 above)
Comment (medgyes) [reply mitzu/webapp/service/user.py:13]: if it's a frozen dataclass then for every email, password update we need to copy the entire object

**Review by IstvanM — APPROVED** [review]

**Pull request merged**